components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java

```diff
@@ -172,4 +213,174 @@ public void testIdentityOAuth2Exception() throws Exception {
         assertFalse(oAuth2ClientValidationResponseDTO.isValidClient());
     }
 
+    @Test
+    public void testIssueAccessToken() throws IdentityException {
+
+        OAuth2AccessTokenRespDTO tokenRespDTO = new OAuth2AccessTokenRespDTO();
+        AccessTokenIssuer accessTokenIssuer = mock(AccessTokenIssuer.class);
+        mockStatic(AccessTokenIssuer.class);
+        when(AccessTokenIssuer.getInstance()).thenReturn(accessTokenIssuer);
+        when(accessTokenIssuer.issue(any(OAuth2AccessTokenReqDTO.class))).thenReturn(tokenRespDTO);
+        assertNotNull(oAuth2Service.issueAccessToken(new OAuth2AccessTokenReqDTO()));
+
+        when(accessTokenIssuer.issue(any(OAuth2AccessTokenReqDTO.class)))
+                .thenThrow(new InvalidOAuthClientException(""));
+        assertEquals(oAuth2Service.issueAccessToken(new OAuth2AccessTokenReqDTO())
+                .getErrorMsg(), "Invalid Client");
+    }
+
+
+    @Test
+    public void testExceptionForIssueAccesstoken() throws IdentityException {
+
+        AccessTokenIssuer accessTokenIssuer = mock(AccessTokenIssuer.class);
+        mockStatic(AccessTokenIssuer.class);
+        when(AccessTokenIssuer.getInstance()).thenReturn(accessTokenIssuer);
+        when(accessTokenIssuer.issue(any(OAuth2AccessTokenReqDTO.class)))
+                .thenThrow(new IdentityOAuth2Exception(""));
+        assertEquals(oAuth2Service.issueAccessToken(new OAuth2AccessTokenReqDTO())
+                .getErrorCode(), "server_error");
+    }
+
+    @Test
+    public void testIsPKCESupportEnabled() {
+
+        OAuthServerConfiguration oAuthServerConfiguration = mock(OAuthServerConfiguration.class);
```

Comment:
you already have a mock object "oAuthServerConfiguration" in [L108](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/584/files#diff-ea8723a20b6b9da1fc22bfba22f89d70R108)

Reply:
okay ill change it